nisystemlink/clients/alarm/_alarm_client.py

```diff
@@ -70,14 +74,38 @@ def create_or_update_alarm(self, request: models.CreateOrUpdateAlarmRequest) ->
         Args:
             request: The request containing alarm_id (user-defined identifier),
                     transition details, and other alarm properties.
+            ignore_conflict: If True, 409 Conflict errors will be ignored and None will be returned.
+                           If False (default), 409 errors will raise an ApiException.
+                           Setting this to True is useful for stateless applications that want to
+                           attempt state transitions without checking the current alarm state first.
 
         Returns:
             The instance_id (unique occurrence identifier) of the created or modified alarm.
             Use this ID for operations like get_alarm(), delete_alarm(), or acknowledge.
+            Returns None if ignore_conflict is True and a 409 Conflict occurs.
 
         Raises:
             ApiException: if unable to communicate with the `/nialarm` Service or provided invalid arguments.
+                A 409 Conflict error occurs when the request does not represent a valid transition
+                for an existing alarm, such as attempting to clear an alarm which is already clear,
+                or attempting to set an alarm which is already set at the given severity level.
+                This error can be suppressed by setting ignore_conflict=True.
         """
+        if ignore_conflict:
```

Comment:
nit: We could remove some duplication by moving this `if` into the `except`: `if ignore_conflict and e.http_status_code == 409:`.